onadata/libs/serializers/xform_serializer.py

```diff
@@ -33,31 +33,40 @@
                                             REPEAT_INDEX_TAGS)
 from onadata.libs.utils.decorators import check_obj
 from onadata.libs.utils.viewer_tools import (
-    enketo_url, get_enketo_preview_url, get_form_url)
+    enketo_url, get_form_url)
 
 
-def _create_enketo_url(request, xform):
+def _create_enketo_urls(request, xform):
     """
-    Generate enketo url for a form
+    Generate enketo urls for a form
 
     :param request:
     :param xform:
-    :return: enketo url
+    :return: enketo urls
     """
     form_url = get_form_url(request, xform.user.username,
                             settings.ENKETO_PROTOCOL, xform_pk=xform.pk,
                             generate_consistent_urls=True)
-    url = ""
-
+    data = {}
     try:
-        url = enketo_url(form_url, xform.id_string)
-        MetaData.enketo_url(xform, url)
+        enketo_urls = enketo_url(form_url, xform.id_string)
+        offline_url = enketo_urls.get("offline_url")
+        MetaData.enketo_url(xform, offline_url)
+        data['offline_url'] = offline_url
+        if 'preview_url' in enketo_urls:
+            preview_url = enketo_urls.get("preview_url")
+            MetaData.enketo_preview_url(xform, preview_url)
+            data['preview_url'] = preview_url
+        if 'single_url' in enketo_urls:
+            single_url = enketo_urls.get("single_url")
+            MetaData.enketo_url(xform, single_url)
```

Comment:
Any reason why we didn't create a static function called `enketo_single_url` in the `meta_data` module like the snippet below?
https://github.com/onaio/onadata/blob/a046b865967baac339094a40f405ae1e629f7400/onadata/apps/main/models/meta_data.py#L257-L265

Reply:
Aaaaah...no reason at all that comes to mind against it.
let me add that